src/gfn/modules.py

```diff
@@ -44,19 +46,31 @@ class GFNModule(ABC, nn.Module):
             been verified.
     """
 
-    def __init__(self, env: Env, module: nn.Module) -> None:
+    def __init__(
+        self,
+        module: nn.Module,
+        preprocessor: Preprocessor | None = None,
+        is_backward: bool = False,
```

Comment:
`DiscretePolicyEstimator` already had a `_forward` attribute. `is_backward` was only part of the `Sampler`. 
So if we wanted a sampler based on a backward discrete policy estimator, we would have needed something like:
```
estimator = DiscretePolicyEstimator(..., forward=False)
sampler = Sampler(estimator, is_backward=True)
```

which is super weird to me, as we would ideally need one flag rather than two.

Are you sure we already had `is_backward` as a generic attribute of modules before removing it ?

Reply:
no now that you put things this way it makes sense to me , thanks. this reason makes sense. as for what came before, we had at some point decided that they should be one level up from the base class - I can't remember why. And since I have no counter argument we should leave it as you have put it, I just found it incongruent with the purpose of the PR so it triggered an alarm in my brain :).

Reply:
ref: https://github.com/saleml/torchgfn/blob/627e90934ae40f9793488c14325cc3199d781eb3/src/gfn/modules.py